### PR TITLE
Downsample queue

### PR DESCRIPTION
**Pull request by movestill**
Add queue for downsample jobs so the user can request a downsample and walk away if another is already running.

Sequence diagram here: https://github.com/jhuapl-boss/boss-manage/blob/downsample-queue/docs/downsample/downsample-sequence-diagram.png

### Related PRs:
https://github.com/jhuapl-boss/boss/pull/63
https://github.com/jhuapl-boss/boss-tools/pull/40

### CODE REVIEW

**Comment by movestill** [comment]
Screen Shot 2020-11-18 at 5 39 38 PM

This is a screenshot of an execution of the updated downsample step function.

File: cloud_formation/configs/api.py [hunk 1]
```diff
@@ -147,11 +148,13 @@ def create_config(bosslet_config, db_config={}):
     # Queue that holds S3 object keys of cuboids to be indexed.
     config.add_sqs_queue(names.index_cuboids_keys.sqs, names.index_cuboids_keys.sqs, 120, 20160)
 
-    #config.add_sqs_queue("DeadLetterQueue", names.deadletter.sqs, 30, 20160) DP XXX
     config.add_sqs_queue(names.deadletter.sqs, names.deadletter.sqs, 30, 20160)
 
+    # ToDo: determine if a dlq needed for this queue.
+    # Downsample jobs.
+    config.add_sqs_queue(names.downsample_queue.sqs, names.downsample_queue.sqs, 300, 20160)
```
Comment (dxenes1) [inline cloud_formation/configs/api.py:155]: What does the 300 second hide timeout do in the case of the downsample queue? Does it mean that if a downsample fails you cant start again for 300 seconds?

**Review by dxenes1 — APPROVED** [review]

File: cloud_formation/configs/api.py (hunk 1 above)
Comment (movestill) [reply cloud_formation/configs/api.py:155]: The message is hidden/unavailable for this amount of time. This prevents another worker from also starting work on the same job. This timeout should be set to longer than it takes to finish the job. If the timeout occurs, this should indicate that the worker crashed, so the message should be made available for other workers.

**Comment by dxenes1** [comment]
Another general question I have, can we start downsamples with base resolutions other than 0 now? I remember this being an issue before but I don't know if it was fixed. For 99% of the cases it doesnt matter but when uploading like a mip3 dataset on cloudvolume it would be nice to conserve that as res3 on bossdb.

**Comment by movestill** [comment]
> Another general question I have, can we start downsamples with base resolutions other than 0 now? I remember this being an issue before but I don't know if it was fixed. For 99% of the cases it doesnt matter but when uploading like a mip3 dataset on cloudvolume it would be nice to conserve that as res3 on bossdb.

I didn’t touch any code related to the base resolution, so not sure.

**Review by perrywilson — APPROVED** [review]

**Comment by perrywilson** [comment]
I read through all of the code. Not sure I understand the heaviside DSL but the changes seem to make sense. I also learned a bit about step functions and SNS vs SQS services.

**Review by sandyhider — APPROVED** [review]

**Pull request merged**